pylint/lint/pylinter.py

```diff
@@ -1028,6 +1027,16 @@ def _discover_files(files_or_modules: Sequence[str]) -> Iterator[str]:
                     if any(root.startswith(s) for s in skip_subtrees):
                         # Skip subtree of already discovered package.
                         continue
+
+                    if _is_ignored_file(
+                        root,
+                        self.config.ignore,
```

Comment:
Ref. https://github.com/PyCQA/pylint/pull/6528#issuecomment-1125734077

This seems to work right @Pierre-Sassoulas?

On `2.13.x` all config is still stored on the `BaseChecker` itself. Since `PyLinter` is a checker you can access it's options with `self.config`. If you need options from another checker you can do `get_global_option` which will look up options in the `config` attribute of all other registered checkers.

So, theoretically this should work and the passing tests seems to confirm this!

Reply:
I've had to do suspicious fix to make it work, in particular use ``ignore-patternS`` instead of  ``ignore-pattern`` in the tests. Did we change an option ?

Reply:
https://github.com/PyCQA/pylint/pull/6563/commits/e211718024f1353f3846586280559386218422b8